src/main/kotlin/org/jetbrains/research/libsl/visitors/ExpressionVisitor.kt

```diff
@@ -211,82 +224,81 @@ class ExpressionVisitor(
     }
 
     override fun visitIntegerNumber(ctx: LibSLParser.IntegerNumberContext): Atomic {
-        if(ctx.suffix() == null) {
-            return IntegerLiteral(
-                ctx.text.toInt(),
-                null,
-                posGetter.getCtxPosition(context.fileName, ctx))
-        } else {
-            return when(ctx.suffix().text) {
-                "b" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toByte(),
-                    "b",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "ub" -> UnsignedInt8Literal(
-                    ctx.text.dropLast(2).toUByte(),
-                    "ub",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "s" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toShort(),
-                    "s",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "us" -> UnsignedInt16Literal(
-                    ctx.text.dropLast(2).toUShort(),
-                    "us",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                null -> IntegerLiteral(
-                    ctx.text.toInt(),
-                    "",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "u" -> UnsignedInt32Literal(
-                    ctx.text.dropLast(1).toUInt(),
-                    "u",
-                    posGetter.getCtxPosition(context.fileName, ctx)
-                )
-                "L" -> IntegerLiteral(
-                    ctx.text.dropLast(1).toLong(),
+        var num = ctx.text.lowercase()
+        return when {
+            num.endsWith("ub") -> UnsignedInt8Literal(
+                ctx.text.dropLast(2).toUByte(),
+                "ub",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+            num.endsWith("b") -> IntegerLiteral(
+                ctx.text.dropLast(1).toByte(),
+                "b",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
+            num.endsWith("us") -> UnsignedInt16Literal(
+                ctx.text.dropLast(2).toUShort(),
+                "us",
+                posGetter.getCtxPosition(context.fileName, ctx)
+            )
```

Comment:
Why not allow `unt16` hex/oct/bin literals (and others) too?

Reply:
Fixed